File: lightyield/old/lightyieldToROOT.cpp
#include <TFile.h>
#include <TH1F.h>
#include <TString.h>
#include <iostream>
#include <fstream>
#include <sstream>

void ToROOT(const char* inputFileName, const char* outputFileName) {
    std::ifstream inputFile(inputFileName);
    if (!inputFile.is_open()) {
        std::cerr << "Error opening input file: " << inputFileName << std::endl;
        return;
    }

    // Read the data and find the maximum channel number
    int maxChannel = 0;
    int channel;
    float count;
    while (inputFile >> channel >> count) {
        if (channel > maxChannel) {
            maxChannel = channel;
        }
    }
    inputFile.clear(); // Clear EOF flag
    inputFile.seekg(0); // Rewind file to read again

    // Create a histogram with a number of bins equal to maxChannel + 1
    TH1F* hist = new TH1F("hist", "Channel Histogram", maxChannel + 1, 0, maxChannel + 1);

    // Fill the histogram with data from the file
    while (inputFile >> channel >> count) {
        hist->SetBinContent(channel + 1, count); // ROOT histograms are 1-based indexing
    }

    // Close the input file
    inputFile.close();

    // Create the output ROOT file
    TFile* outFile = TFile::Open(outputFileName, "RECREATE");
    if (!outFile || outFile->IsZombie()) {
        std::cerr << "Error creating output file: " << outputFileName << std::endl;
        return;
    }

    // Write the histogram to the file
    hist->Write();
    outFile->Close();

    std::cout << "Histogram saved to " << outputFileName << std::endl;

    // Cleanup
    delete hist;
}
void lightyieldToROOT(){
    Char_t filename[100]; // temp store 
    Char_t output[100]; // temp store 
    for(int i = 0; i< 15; i++){
        sprintf(filename, "BPS_CH%i.his.txt", i);
        sprintf(output, "BPS_CH%i.root", i);
        ToROOT(filename, output);
    }
}

File: lightyield/old/lightyieldToROOT.py
import ROOT
import sys
from array import array

def lightyieldToROOT(input_file, output_file):
    # Create a new ROOT file
    root_file = ROOT.TFile(output_file, "RECREATE")
    
    # Create a ROOT tree
    tree = ROOT.TTree("tree", "Histogram data")

    # Variable to hold the channel number
    channel = array('i', [0])

    # Create a branch in the tree
    tree.Branch("channel", channel, "channel/I")

    # Open the input text file
    with open(input_file, 'r') as f:
        for line in f:
            # Split each line into channel number and count
            parts = line.split()
            if len(parts) == 2:
                channel_nb = int(parts[0])
                count_nb = int(parts[1])
                
                # Fill the branch with the channel number, repeating it `count_nb` times
                for _ in range(count_nb):
                    channel[0] = channel_nb
                    tree.Fill()

    # Write the tree to the ROOT file and close the file
    tree.Write()
    root_file.Close()
    
def main():
   for i in range(0, 15):
        filename = f'BPS_CH{i}.his.txt'
        lightyieldToROOT(filename, filename[:-8]+".root")

if __name__ == "__main__":
    main()

